extras/subscript.go

```diff
@@ -0,0 +1,115 @@
+package extras
+
+import (
+	"github.com/gohugoio/hugo-goldmark-extensions/extras/ast"
+	"github.com/yuin/goldmark"
+	gast "github.com/yuin/goldmark/ast"
+	"github.com/yuin/goldmark/parser"
+	"github.com/yuin/goldmark/renderer"
+	"github.com/yuin/goldmark/renderer/html"
+	"github.com/yuin/goldmark/text"
+	"github.com/yuin/goldmark/util"
+)
+
+type subscriptDelimiterProcessor struct {
+}
+
+func (p *subscriptDelimiterProcessor) IsDelimiter(b byte) bool {
+	return b == '~'
+}
+
+func (p *subscriptDelimiterProcessor) CanOpenCloser(opener, closer *parser.Delimiter) bool {
+	return opener.Char == closer.Char
+}
+
+func (p *subscriptDelimiterProcessor) OnMatch(consumes int) gast.Node {
+	return ast.NewSubscript()
+}
+
+var defaultSubscriptDelimiterProcessor = &subscriptDelimiterProcessor{}
+
+type subscriptParser struct {
+}
+
+var defaultSubscriptParser = &subscriptParser{}
+
+// NewSubscriptParser returns a new InlineParser that parses subscript expressions.
+func NewSubscriptParser() parser.InlineParser {
```

Comment:
This is related to the above (one parser), but I suggest that we for starters unexports everything that's not related to getting a extension.